include/nbl/builtin/glsl/ext/FFT/fft.glsl

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_GLSL_EXT_FFT_INCLUDED_
+#define _NBL_GLSL_EXT_FFT_INCLUDED_
+
+#include <nbl/builtin/glsl/math/complex.glsl>
+
+// Shared Memory
+#include <nbl/builtin/glsl/workgroup/shared_arithmetic.glsl>
+
+
+#ifndef _NBL_GLSL_EXT_FFT_MAX_DIM_SIZE_
+#error "_NBL_GLSL_EXT_FFT_MAX_DIM_SIZE_ should be defined."
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_MAX_ITEMS_PER_THREAD
+#error "_NBL_GLSL_EXT_FFT_MAX_ITEMS_PER_THREAD should be defined."
+#endif
+
+
+// TODO: Investigate why +1 solves all glitches
+#define _NBL_GLSL_EXT_FFT_SHARED_SIZE_NEEDED_ _NBL_GLSL_EXT_FFT_MAX_DIM_SIZE_ + 1
```

Comment:
ah yeah we still have the issue of smem usage left to address... right?

Reply:
Yep

Reply:
ok we should cook something up for that for the next review.